diff --git a/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/java/com/incomingcall/IncomingCallModule.java b/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/java/com/incomingcall/IncomingCallModule.java
index 5989ad0..abef6a1 100644
--- a/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/java/com/incomingcall/IncomingCallModule.java
+++ b/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/java/com/incomingcall/IncomingCallModule.java
@@ -13,6 +13,7 @@ import com.facebook.react.bridge.Callback;
 import com.facebook.react.bridge.Promise;
 import com.facebook.react.bridge.WritableMap;
 import com.facebook.react.bridge.WritableNativeMap;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
 
 public class IncomingCallModule extends ReactContextBaseJavaModule {
 
@@ -57,6 +58,14 @@ public class IncomingCallModule extends ReactContextBaseJavaModule {
         }
     }
 
+    @ReactMethod
+    public void dismiss() {
+        if (UnlockScreenActivity.active) {
+            UnlockScreenActivity.dismiss();
+        }
+        return;
+    }
+
     private Context getAppContext() {
         return this.reactContext.getApplicationContext();
     }
diff --git a/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/java/com/incomingcall/UnlockScreenActivity.java b/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/java/com/incomingcall/UnlockScreenActivity.java
index 5d1d0aa..a18e590 100644
--- a/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/java/com/incomingcall/UnlockScreenActivity.java
+++ b/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/java/com/incomingcall/UnlockScreenActivity.java
@@ -11,8 +11,10 @@ import android.widget.TextView;
 import android.view.View;
 import android.net.Uri;
 import android.os.Vibrator;
+import android.os.VibrationEffect;
 import android.content.Context;
 import android.media.MediaPlayer;
+import android.media.AudioAttributes;
 import android.provider.Settings;
 import java.util.List;
 import java.util.Timer;
@@ -111,7 +113,19 @@ public class UnlockScreenActivity extends AppCompatActivity implements UnlockScr
         getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN | WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON | WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
                 | WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED | WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD);
 
-        v.vibrate(pattern, 0);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            AudioAttributes audioAttributes = new AudioAttributes.Builder()
+                .setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
+                .setUsage(AudioAttributes.USAGE_ALARM)
+                .build();
+            VibrationEffect ve = VibrationEffect.createWaveform(
+                pattern,
+                0
+            );
+            v.vibrate(ve, audioAttributes);    
+        } else {
+            v.vibrate(pattern, 0);
+        }
         player.start();
 
         AnimateImage acceptCallBtn = findViewById(R.id.ivAcceptCall);
@@ -157,6 +171,15 @@ public class UnlockScreenActivity extends AppCompatActivity implements UnlockScr
         dismissDialing();
     }
 
+    public static void dismiss() {
+        WritableMap params = Arguments.createMap();
+        v.cancel();
+        player.stop();
+        player.prepareAsync();
+        IncomingCallModule.reactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit("endCall", params);
+        fa.finish();
+    }
+
     private void acceptDialing() {
         WritableMap params = Arguments.createMap();
         params.putBoolean("accept", true);
diff --git a/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/res/values/colors.xml b/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/res/values/colors.xml
index 104b062..c580c9c 100644
--- a/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/res/values/colors.xml
+++ b/node_modules/@bob.hardcoder/react-native-incoming-call/android/src/main/res/values/colors.xml
@@ -1,3 +1,3 @@
 <resources>
-    <color name="background">#141C2E</color>
+    <color name="background">#000000</color>
 </resources>
